### PR TITLE
Add e2ectl logs export command

### PR DESCRIPTION
**Pull request by rossf7**
Towards https://github.com/giantswarm/giantswarm/issues/4606

```bash
e2ectl logs export -h
Export logs.

Usage:
  e2ectl logs export [flags]

Flags:
  -h, --help              help for export
      --logs-dir string   Logs are exported to this directory.
      --name string       Name of e2e cluster. (default "kind")
```

Based on the [kind command]. Only difference is the logs dir must be specified. kind creates a [temp dir] but extra hoops are needed to make this play nicely with Docker. 

[kind command]: https://github.com/kubernetes-sigs/kind/blob/master/cmd/kind/export/logs/logs.go
[temp dir]: https://github.com/kubernetes-sigs/kind/blob/543c84e5b90cc20c4daec819827e32e95e428793/pkg/fs/fs.go#L31

### CODE REVIEW

File: cmd/cluster/delete/runner.go [hunk 1]
```diff
@@ -3,18 +3,13 @@ package delete
...
 	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/micrologger"
 	"github.com/spf13/cobra"
 	"sigs.k8s.io/kind/pkg/cluster"
 )
 
-const (
-	waitForReady = 2 * time.Minute
-)
-
```
Comment (rossf7) [inline cmd/cluster/delete/runner.go]: Removing dead code.

File: cmd/logs/export/runner.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	}
+
+	err = r.run(ctx, cmd, args)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
```
Comment (rossf7) [inline cmd/logs/export/runner.go]: This is the important bit. The rest is wiring for the commands.

**Comment by rossf7** [comment]
I worked on this during sig-releng SWH. Next step is using in managed apps e2e tests.

File: cmd/logs/export/runner.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+func (r *runner) run(ctx context.Context, cmd *cobra.Command, args []string) error {
+	var err error
+
+	var known bool
```
Comment (MarcelMue) [inline cmd/logs/export/runner.go]: Why define the variable here?
`known, err := cluster.IsKnown(r.flag.Name)` should do the job imo

**Review by MarcelMue — APPROVED** [review]

File: cmd/logs/export/runner.go (hunk 3 above)
Comment (rossf7) [reply cmd/logs/export/runner.go]: It was copied from the other commands. But yes that is simpler. Done.

**Review by sslavic — COMMENTED** [review]
LGTM

**Pull request merged**